### PR TITLE
Update to Version 3 currently `next` tag

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Relates to: https://github.com/SAP/cloud-sdk-backlog/issues/1017

New [v2-main](https://github.com/SAP-samples/cloud-sdk-js/tree/v2-main) branch.
Note there is a script to update version in the repo after the release.

### CODE REVIEW

File: samples/cds-sample-application/package.json [hunk 1]
```diff
@@ -25,26 +25,26 @@
...
-    "generate-client": "npx generate-odata-client --inputDir resources/service-specs --outputDir src/generated --clearOutputDir --generateJs=false --serviceMapping=resources/service-specs/service-mapping.json"
+    "generate-client": "npx generate-odata-client --input resources/service-specs --outputDir src/generated --clearOutputDir --transpile=false --optionsPerService=resources/service-specs/service-mapping.json"
   },
   "dependencies": {
     "@nestjs/common": "^8.4.7",
     "@nestjs/core": "^8.4.7",
     "@nestjs/platform-express": "^8.4.7",
-    "@sap/cds": "^6.1.3",
-    "@sap/cds-dk": "^6.5.2",
+    "@sap/cds": "6.1.3",
```
Comment (tomfrenken) [inline samples/cds-sample-application/package.json:34]: Why are these dependencies pinned here?

File: samples/cds-sample-application/package.json [hunk 2]
```diff
@@ -25,26 +25,26 @@
     "ci-package": "rimraf deployment && mkdir deployment && cp -r dist package.json package-lock.json db srv .cdsrc.json deployment/",
     "cds-build": "cds build/all",
     "cds-deploy": "cds deploy",
-    "generate-client": "npx generate-odata-client --inputDir resources/service-specs --outputDir src/generated --clearOutputDir --generateJs=false --serviceMapping=resources/service-specs/service-mapping.json"
+    "generate-client": "npx generate-odata-client --input resources/service-specs --outputDir src/generated --clearOutputDir --transpile=false --optionsPerService=resources/service-specs/service-mapping.json"
```
Comment (tomfrenken) [inline samples/cds-sample-application/package.json:28]: I think here we should use `options-per-service.json` instead

**Review by tomfrenken — APPROVED** [review]
LGTM, however the script which is changing the sdk version won't work for the samples:
  'samples/cf-sample-application',
  'samples/helm-sample-application/application',
  'samples/k8s-sample-application',

as the package.json will be updated by the remote repository

**Comment by FrankEssenberger** [comment]
> LGTM, however the script which is changing the sdk version won't work for the samples: 'samples/cf-sample-application', 'samples/helm-sample-application/application', 'samples/k8s-sample-application',
> 
> as the package.json will be updated by the remote repository

I already updated the package.json in the repo from where the package.json files are coming form.

File: samples/cds-sample-application/package.json (hunk 1 above)
Comment (FrankEssenberger) [reply samples/cds-sample-application/package.json:34]: I had an issue with broken types in the `cds` dependency. I checked which was the installed version which was working and pinned it to this version.

File: samples/cds-sample-application/package.json (hunk 2 above)
Comment (FrankEssenberger) [reply samples/cds-sample-application/package.json]: yes

**Pull request merged**